java/src/main/java/com/google/appengine/tools/pipeline/di/JobRunServiceComponentContainer.java

```diff
@@ -0,0 +1,17 @@
+package com.google.appengine.tools.pipeline.di;
+
+
+
+public final class JobRunServiceComponentContainer {
+  private JobRunServiceComponentContainer() {}
+
+  // suggested by chat-gpt to get static thread-safe initialization with min synchronization overhead
+  private static class Holder {
+    private static final JobRunServiceComponent INSTANCE =
+      DaggerJobRunServiceComponent.create();
```

Comment:
q: if `.create` creates a new fresh container even if there is `@Singleton`; can we face issues with this approach? As here container will mostly live during all app running, not sure if that could be a problem (no idea if it is OK, or container should be short lived, etc specially if we are having memory issues)

Reply:
yeah... that's the point; it's a problem currently, bc that was done in `init()` of every servlet, so ending up with multiple Component instances when we expected singleton